src/main/java/com/artipie/asto/Simple.java

```diff
@@ -0,0 +1,157 @@
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.asto;
+
+import com.jcabi.log.Logger;
+import hu.akarnokd.rxjava3.jdk8interop.CompletableInterop;
+import hu.akarnokd.rxjava3.jdk8interop.SingleInterop;
+import io.reactivex.rxjava3.core.Completable;
+import io.reactivex.rxjava3.core.Flowable;
+import io.reactivex.rxjava3.core.Single;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Flow;
+import java.util.stream.Collectors;
+import org.reactivestreams.FlowAdapters;
+
+/**
+ * Simple storage, in files.
+ *
+ * @since 0.1
+ */
+public final class Simple implements Storage {
+
+    /**
+     * Where we keep the data.
+     */
+    private final Path dir;
+
+    /**
+     * Ctor.
+     *
+     * @throws IOException If fails
+     */
+    public Simple() throws IOException {
+        this(Files.createTempDirectory("asto"));
+    }
+
+    /**
+     * Ctor.
+     *
+     * @param path The path to the dir
+     */
+    public Simple(final Path path) {
+        this.dir = path;
+    }
+
+    @Override
+    public CompletableFuture<Boolean> exists(final String key) {
+        return (CompletableFuture<Boolean>) Single.fromCallable(
+            () -> {
+                final Path path = Paths.get(this.dir.toString(), key);
+                return Files.exists(path);
+            }).to(SingleInterop.get());
+    }
+
+    @Override
+    public CompletableFuture<Collection<String>> list(final String prefix) {
+        return (CompletableFuture<Collection<String>>) Single.fromCallable(
+            () -> {
+                if (!prefix.endsWith("/")) {
+                    throw new IllegalArgumentException(
+                        String.format(
+                            "The prefix must end with a slash: \"%s\"",
+                            prefix
+                        )
+                    );
+                }
+                final Path path = Paths.get(this.dir.toString(), prefix);
+                final Collection<String> keys = Files.walk(path)
+                    .filter(Files::isRegularFile)
+                    .map(Path::toString)
+                    .map(
+                        p -> p.substring(
```

Comment:
@Sammers21 what does this code do? I'd move it to new method with readable name and javadoc